solvation_analysis/rdf_parser.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from scipy.interpolate import interp1d, UnivariateSpline
+import scipy
+import matplotlib.pyplot as plt
+import warnings
+
+
+def interpolate_rdf(bins, rdf, floor=0.05, cutoff=5):
+    start = np.argmax(rdf > floor)  # will return first index > rdf
+    end = np.argmax(bins > cutoff)  # will return first index > cutoff
+    bounds = (bins[start], bins[end - 1])
+    f = UnivariateSpline(bins[start:end], rdf[start:end], k=4, s=0)
+    return f, bounds
+
+
+def identify_minima(f):
+    assert isinstance(
+        f, scipy.interpolate.fitpack2.InterpolatedUnivariateSpline
+    ), "this function is compatible with the scipy.interpolate package"
```

Comment:
This error message gets outputted if the function isn't a instance, is this sufficiently cleaer for users to understand what's happening?

Also it's more lines, but I'd suggest avoiding the assert here and going for an if/else with a raise. This will let you set the raise to an error that is more sensible (AssertionError is not what folks will be expecting, maybe a TypeError?)

Reply:
I'd sooner just accept anything, put it inside a try/except and complain later. It'd be annoying if I were interpolating differently and this function refused to work for me.

Reply:
Got it, I threw all this inside a try catch that should catch anything with a .derivative() method and give a slightly more informative error. Let me know if the new solution looks alright.